### PR TITLE
feat(frontend): new send modal for all tokens

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

We include the possibility to send tokens from the main page. To do that, we first create a modal that "merge" both flows for IC and ETH tokens.
The modal will display only the enabled tokens that have non-null balance.

# Changes

- Create new component `SendTokensList` with the desired list of tokens:
   - PR #1887 
- Create `SendModal` with the usage of both IC and ETH send modals:
  - PR #1886 
  - PR #1907 
- Create `Send` button.

# Tests

I sent out ICP and ETH on local replica and it worked as expected, as usual.

### CODE REVIEW

**Review by peterpeterparker — COMMENTED** [review]
Can you split this PR in smaller PRs?

**Comment by AntonioVentilii-DFINITY** [comment]
Sure! I will list here the PRs:

**Comment by AntonioVentilii-DFINITY** [comment]
@peterpeterparker this is finally ready for review

File: src/frontend/src/lib/components/hero/Actions.svelte [hunk 1]
```diff
@@ -18,6 +18,7 @@
 	import { tokenWithFallback } from '$lib/derived/token.derived';
 	import Receive from '$lib/components/receive/Receive.svelte';
 	import ContextMenu from '$lib/components/hero/ContextMenu.svelte';
+	import Send from '$lib/components/send/Send.svelte';
 
 	export let send = false;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/hero/Actions.svelte:23]: we still need this option then?

File: src/frontend/src/lib/components/send/Send.svelte [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	import { modalSend } from '$lib/derived/modal.derived';
+	import SendButton from '$lib/components/send/SendButton.svelte';
+	import SendModal from '$lib/components/send/SendModal.svelte';
+
+	const modalId = Symbol();
+</script>
+
+<SendButton on:click={() => modalStore.openSend(modalId)} />
+
+{#if $modalSend}
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/send/Send.svelte:12]: shouldn't you check the modalId here?

File: src/frontend/src/lib/components/hero/Actions.svelte (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/hero/Actions.svelte:23]: nope, but I was going to remove it in another PR

File: src/frontend/src/lib/components/send/Send.svelte (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/send/Send.svelte]: right! very sorry! I forgot it! included in the last commit

File: src/frontend/src/lib/components/hero/Actions.svelte (hunk 1 above)
Comment (peterpeterparker) [reply src/frontend/src/lib/components/hero/Actions.svelte:23]: Ok. Please add a TODO.

File: src/frontend/src/lib/components/send/Send.svelte [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	import { modalSend } from '$lib/derived/modal.derived';
+	import SendButton from '$lib/components/send/SendButton.svelte';
+	import SendModal from '$lib/components/send/SendModal.svelte';
+
+	const modalId = Symbol();
+</script>
+
+<SendButton on:click={() => modalStore.openSend(modalId)} />
+
+{#if $modalSend && $modalStore?.data === modalId}
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/send/Send.svelte:12]: Unrelated to this PR, thinking out loud, maybe we can create a generic component?

```
<script>
   export let open = false;

   const modalId = Symbol();
<script>

{#if open && $modalStore?.data === modalId}
  <slot />
{/if}

// Used

<ModalGuard open={$modalSend} />
   <SendModal on:nnsClose />
</ModalGuard>
```

Maybe the modal can even be passed as a component?

```
<ModalSomething open={$modalSend} cmp={SendModal} />
```

Again, just thinking out loud. Sharing if interesting.

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks

File: src/frontend/src/lib/components/hero/Actions.svelte (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/hero/Actions.svelte:24]: Done:

- PR #1951

File: src/frontend/src/lib/components/send/Send.svelte (hunk 3 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/send/Send.svelte:12]: Done with:

- PR #1978
- PR #1979

**Pull request merged**